charts/komodor-agent/templates/daemonset.yaml

```diff
@@ -27,7 +27,10 @@ spec:
       affinity:
         {{- toYaml .Values.components.komodorDaemon.affinity | nindent 8 }}
       nodeSelector:
+        kubernetes.io/os: linux
+        {{- if not (empty (((.Values.components).komodorDaemon).nodeSelector)) }}
```

Comment:
why this if?

Reply:
Without if it will add {}.

Therefore it would be:
nodeSelector:
   kubernetes.io/os: linux
   {}
   
   And this will fail